programs/wen-new-standard/src/instructions/mint/royalties/modify.rs

```diff
@@ -0,0 +1,160 @@
+        invoke(
+            &remove_key(
+                &self.token_program.key(),
+                &self.mint.key(),
+                &self.authority.key(),
+                field,
+                false,
+            ),
+            &[
+                self.mint.to_account_info(),
+                self.authority.to_account_info(),
+            ],
+        )?;
+
+        Ok(())
+    }
+}
+
+pub fn handler(ctx: Context<ModifyRoyalties>, args: ModifyRoyaltiesArgs) -> Result<()> {
+    let mint_account = ctx.accounts.mint.to_account_info().clone();
+    let mint_account_data = mint_account.try_borrow_mut_data()?;
+    let mint_data = StateWithExtensions::<BaseStateMint>::unpack(&mint_account_data)?;
+    let metadata = mint_data.get_variable_len_extension::<TokenMetadata>()?;
+
+    drop(mint_account_data);
+
+    let creators_arg = &args.creators;
+
+    // validate that the fee_basis_point is less than 10000 (100%)
+    require!(
+        args.royalty_basis_points <= 10000,
+        MetadataErrors::RoyaltyBasisPointsInvalid
+    );
+
+    // since this field is already there, it will just update it with the new value if there is one
+    ctx.accounts.update_token_metadata_field(
+        Field::Key(ROYALTY_BASIS_POINTS_FIELD.to_owned()),
+        args.royalty_basis_points.to_string(),
+    )?;
+
+    let mut total_share: u8 = 0;
+    // add creators and their respective shares to metadata
+    for creator in creators_arg {
+        // validate that the creator is a valid publickey
+        match Pubkey::from_str(&creator.address) {
+            Ok(_) => {
+                total_share = total_share
+                    .checked_add(creator.share)
+                    .ok_or(MetadataErrors::CreatorShareInvalid)?;
+                ctx.accounts.update_token_metadata_field(
+                    Field::Key(creator.address.clone()),
+                    creator.share.to_string(),
+                )?;
+            }
+            Err(_) => return Err(MetadataErrors::CreatorAddressInvalid.into()),
+        }
+    }
+
+    if total_share != 100 {
+        return Err(MetadataErrors::CreatorShareInvalid.into());
+    }
+
+    // remove all the creators that don't need to be there anymore
+    let keys_to_remove: Vec<String> = metadata
+        .additional_metadata
+        .iter()
+        .filter_map(|(key, _)| match Pubkey::from_str(key) {
+            Ok(pubkey) => {
+                if !creators_arg
+                    .iter()
+                    .any(|creator| creator.address == pubkey.to_string())
+                {
+                    Some(key.clone())
+                } else {
+                    None
+                }
+            }
+            Err(_) => None,
+        })
+        .collect();
+
+    for key in keys_to_remove {
+        ctx.accounts.remove_token_metadata_field(key)?;
+    }
+
+    // transfer minimum rent to mint account
+    update_account_lamports_to_minimum_balance2(
```

Comment:
What exemption costs are we expecting to change here?

Reply:
Since we could add additional creator, we should have the check in place that increase space if needed. Or it will break everything